sam/models/base_model.py

```diff
@@ -314,62 +235,22 @@ def preprocess_fit(
         self.prediction_cols_ += ["predict_lead_{}_mean".format(p) for p in self.predict_ahead]
         self.n_outputs_ = len(self.prediction_cols_)
 
-        # Remove the first n rows because they are nan anyway because of rolling features
-        if len(self.rolling_window_size) > 0:
-            X_transformed = X_transformed.iloc[max(self.rolling_window_size) :]
-            y_transformed = y_transformed.iloc[max(self.rolling_window_size) :]
-        # Filter rows where the target is unknown
-        X_transformed = X_transformed.loc[~targetnanrows]
-        y_transformed = y_transformed.loc[~targetnanrows]
-
-        assert_contains_nans(
-            X_transformed, "Data cannot contain nans. Imputation not supported for now"
-        )
+        X_trans, y_trans = self.preprocess(X, y, train=True)
+        X_trans, y_trans = remove_until_first_value(X_trans, y_trans)
```

Comment:
Shouldn't this also happen when predicting? Maybe I missed it, but I don't think we're removing the NaNs when predicting

Reply:
I think we never did that.

When training: nan -> error, so we remove them

When predicting: nan -> nan (I think). I agree that can be tricky, so a safe guard could work (new issue? 😁 )

Reply:
Ah, I think you're right, as long as the prediction doesn't fail, but it can of course contain NaNs